### DISCUSSION TITLE
What actions are triggered by reserving a CIDR range in IPAM?

### DISCUSSION THREAD

**Discussion by serottez** [Azure/ipam#323] [Q&A]
Hi, I am wondering what exactly the IPAM application does when a user reserves a CIDR range? From my understanding, IPAM only has read permissions on Azure resources (excluding the Azure resources that were deployed by IPAM). So, when a user reserves a CIDR range in IPAM, does IPAM only add a record to the Cosmos DB database, or does a reservation trigger other actions/updates, too? Thank you!

**Comment by DCMattyG** [comment]
Hi @serottez, when a **Reservation** is created it does indeed update the data within CosmosDB. After creating a **Reservation**, you will be provided with an **ID** which you will need to apply as a Tag on the new **Virtual Network** you create within Azure using the provided CIDR so that the Azure IPAM tool will know which **Virtual Network** to associate with the **Reservation** and fulfill it.

**Comment by georgegil** [comment]
This can certainly do with being documented better imho. 

Took me a bit to work it out

**Reply by DCMattyG** [reply]
I 100% concur with you @georgegil, and my apologies.

I am indeed working on some highly-improved documentation around this process which should be ready shortly.

**Reply by lansalot** [reply]
Hi, any idea on documentation timelines? The solution is woefully under-documented, and the API section shows a fraction of what's available.

I went round and round trying to work out how to create a reservation before realising I couldn't do it in the portal - which seems like a huge omission; I can generate a block but I can't reserve it?

Sections on how implementing with mature environments (multiple subnets, regions, and on-premise) to document what's already there and not create overlaps would be ideal.

**Reply by DCMattyG** [reply]
Hi @lansalot, I'm actually working on this as we speak. Take a look at the new section I added in the v3.7.0 branch at the bottom of the How-To README.

https://github.com/Azure/ipam/blob/ipam-3.7.0/docs/how-to/README.md

I'm hoping to have all of this out the door by the end of February at the latest, just working on a bit more testing of all the latest changes and need to add some screenshots to the new docs.

Let me know what you think so far!

**Reply by lansalot** [reply]
Thanks, will take a look 

Also, I just found out a reservation can now be made in the portal, so that's a bit more useful. 

A

**Reply by DCMattyG** [reply]
Hahaha, yes....I have WAY more documentation to add. I was more focused on getting the features in place and now I'll be spending some time circling back and making sure all of them are properly documented.

**Reply by lansalot** [reply]
Thanks, that looks a bit better. The mention of ellipses I presume will change in future release as at the moment it's typically a down-arrow instead.

**Reply by lansalot** [reply]
(Noted it does still say Reservations have to be done in the API - they can be done in the portal in 3.6.0)

**Comment by serottez** [comment]
Thank you @DCMattyG. Is the CosmosDB record the only modification made when a CIDR range is reserved? I am considering implementing this in a preexisting and mature Azure environment, where multiple end users will have permission to make CIDR reservations. I want to make sure that reservations made by these users do not have domino effects on other Azure resources. Thank you!

**Reply by DCMattyG** [reply]
Indeed you are good @serottez. This solution was designed for multiple users to share and perform exactly the process you outlined.

And as always, Azure IPAM will never modify any Azure resources as it only has **Reader** permissions to the Root Management Group.
